src/main/java/org/elasticsearch/gateway/Gateway.java

```diff
@@ -159,7 +159,7 @@ public void performStateRecovery(final GatewayStateRecoveredListener listener) t
         }
         ClusterState.Builder builder = ClusterState.builder(clusterName);
         builder.metaData(metaDataBuilder);
-        listener.onSuccess(builder.build());
+        listener.onSuccess(builder.randomUuid().build());
```

Comment:
I think we should have a randomUUID by default and only if somebody specify one explicitly  we should use it. ie. remove the method entirely?